token-client-spring/src/main/java/no/nav/security/token/support/client/spring/oauth2/Oauth2ClientRequestInterceptor.java

```diff
@@ -0,0 +1,52 @@
+package no.nav.security.token.support.client.spring.oauth2;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+import no.nav.security.token.support.client.core.oauth2.OAuth2AccessTokenService;
+import no.nav.security.token.support.client.spring.ClientConfigurationProperties;
+
+/**
+ *
+ * Interceptor som veksler inn et token og setter Authorization header til
+ * dette, med audience kun for target app. Gjeldende klient konfigurasjon kan
+ * slås opp via en konfigurerbar matcher. Må settes opp av de enkelte
+ * klient-applikasjonene selv, iingen automatisk registrering i application
+ * context. {@link ClientConfigurationPropertiesMatcher}
+ *
+ */
+public class Oauth2ClientRequestInterceptor implements ClientHttpRequestInterceptor {
```

Comment:
Maybe rename class to `OAuth2ClientRequestInterceptor` in order to match the overall naming strategy